packages/backend/src/open_payments/payment/outgoing/service.ts

```diff
@@ -419,10 +419,10 @@ async function addSentAmount(
 ): Promise<OutgoingPayment> {
   const fundingZeroOrUndefined =
     payment.state === OutgoingPaymentState.Funding ? BigInt(0) : undefined
-  let sent = value || (await deps.accountingService.getTotalSent(payment.id))
```

Comment:
_This is a guess about the desired behaviour_

the main issue here was using `||` instead of `??`. When we call this function just above:
https://github.com/interledger/rafiki/blob/cbbc41613fc020ee5c2e6d974e7fc95755dec408/packages/backend/src/open_payments/payment/outgoing/service.ts#L160

we pass in `value: BigInt(0)` . `BigInt(0)` is falsy, so we end up having calling `deps.accountingService.getTotalSent(payment.id))` when we made value a defined value.